src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -66,18 +73,28 @@ object CromwellRuntimeAttributes {
     }
   }
 
-  def unsupportedKeys(keys: Iterable[String], backendType: BackendType): Option[String] = {
-    val supportedKeys = backendType.supportedKeys map { _.key }
-    val unsupportedKeys = keys.toSet -- supportedKeys
-
+  def logMessageForUnsupportedKeys(unsupportedKeys: Set[String], backendType: BackendType) = {
     if (unsupportedKeys.isEmpty) None
     else Option(s"Found unsupported keys for backend '$backendType': " + unsupportedKeys.toSeq.sorted.mkString(", "))
   }
 
+  def unsupportedKeys(keys: Iterable[String], backendType: BackendType): Set[String] = {
+    val supportedKeys = backendType.supportedKeys map { _.key }
+    keys.toSet -- supportedKeys
+  }
+
   val LocalDiskName = "local-disk"
   val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, sizeGb=10)
 
-  private val keys = Set("cpu", "disks", "docker", "zones", "continueOnReturnCode", "failOnStderr", "preemptible", "memory")
+  private case class ValidKeyType(key: String, validTypes: Set[WdlType] = Set(WdlStringType))
+  private val keys = Set(ValidKeyType("cpu", Set(WdlIntegerType, WdlStringType)),
+                        ValidKeyType("disks"),
+                        ValidKeyType("docker"),
+                        ValidKeyType("zones"),
+                        ValidKeyType("continueOnReturnCode", Set(WdlBooleanType, WdlArrayType(WdlIntegerType))),
+                        ValidKeyType("failOnStderr", Set(WdlBooleanType, WdlStringType)),
+                        ValidKeyType("preemptible", Set(WdlIntegerType, WdlStringType)),
```

Comment:
I think this should just be integer. The strings will get coerced down if appropriate?